GitCommands/PathUtil.cs

```diff
@@ -78,19 +78,19 @@ public static bool IsLocalFile(string fileName)
         /// <param name="path">A path to check.</param>
         /// <returns><see langword="true"/> if the given path starts with 'http', 'ssh' or 'git'; otherwise <see langword="false"/>.</returns>
         [Pure]
-        public static bool IsUrl(string path)
+        public static bool IsUrl(string? path)
         {
-            return !string.IsNullOrEmpty(path)
+            return !Strings.IsNullOrEmpty(path)
                 && (path.StartsWith("http:", StringComparison.CurrentCultureIgnoreCase)
                  || path.StartsWith("https:", StringComparison.CurrentCultureIgnoreCase)
                  || path.StartsWith("git:", StringComparison.CurrentCultureIgnoreCase)
                  || path.StartsWith("ssh:", StringComparison.CurrentCultureIgnoreCase)
                  || path.StartsWith("file:", StringComparison.CurrentCultureIgnoreCase));
         }
 
-        public static bool CanBeGitURL(string url)
+        public static bool CanBeGitURL(string? url)
         {
-            if (string.IsNullOrWhiteSpace(url))
+            if (Strings.IsNullOrWhiteSpace(url))
```

Comment:
Thanks, pushed a commit to address all instances of `== null` and `!= null` in the core source.

It makes no difference to null analysis. It's mostly style. Academically, it can make a difference if a type overrides the `==` operator to do something suprising, which I've never seen in practice.

I don't mind the style change, but there's a lot of muscle memory here. Ideally we find an analyzer to enforce this to reduce friction.